MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js

```diff
@@ -71,6 +71,19 @@ define([
             }
         },
 
+        /**
+         * Wait for component to initialize
+         */
+        waitForComponent: function (callback, component) {
```

Comment:
We already have a method that performs similar action on this component (waitForContainer), I would suggest we extract the if condition and define it as a parameter, rename the method to waitForCondition and use it for both approaches.

Something like this:
```
waitForCondition: function (condition, callback) {
            if (condition()) {
                setTimeout(function () {
                    this.waitForCondition(condition, callback);
                }.bind(this), 100);
            } else {
                callback();
            }
        },
```

And to call that method we would do:

```
            this.waitForCondition(function () {
                return $(this.directoryTreeSelector).length === 0
            }.bind(this), function () {
                this.getJsonTree();
                this.initEvents();
                this.checkChipFiltersState();
            }.bind(this));
```

Reply:
completely agree with you.